core/templates/components/on-screen-keyboard/on-screen-keyboard.component.html

```diff
@@ -0,0 +1,226 @@
+<div class="oppia-osk-container" ng-if="$ctrl.showOSK()" ng-click="$ctrl.activateGuppy()">
+  <div class="oppia-osk-navbar">
+    <button ng-class="($ctrl.currentTab === 'mainTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('mainTab')">
+      123
+    </button>
+    <button ng-class="($ctrl.currentTab === 'functionsTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('functionsTab')">
+      f(x)
+    </button>
+    <button ng-if="!$ctrl.customizable" ng-class="($ctrl.currentTab === 'lettersTab') ? 'oppia-osk-btn-tab-active' : 'oppia-osk-btn-tab-inactive'" ng-click="$ctrl.changeTab('lettersTab')">
+      abc
+    </button>
+    <button class="oppia-osk-btn-hide" ng-click="$ctrl.hideOSK()">
+      HIDE <i class="fa fa-angle-down"></i>
+    </button>
+
+  </div>
+  <div class="oppia-osk-custom-letters-div" ng-if="$ctrl.customizable && $ctrl.currentTab === 'mainTab'">
+    <button class="oppia-osk-btn-letter" ng-click="$ctrl.insertString(letter)" ng-repeat="letter in $ctrl.customLetters"><[letter]></button>
+  </div>
+  <div class="oppia-osk-main-tab" ng-if="$ctrl.currentTab==='mainTab'">
+    <button class="oppia-osk-btn-number number-7" ng-click="$ctrl.insertString('7')">7</button>
+    <button class="oppia-osk-btn-number number-8" ng-click="$ctrl.insertString('8')">8</button>
+    <button class="oppia-osk-btn-number number-9" ng-click="$ctrl.insertString('9')">9</button>
+    <button class="oppia-osk-btn-operator" ng-click="$ctrl.insertString('+')">+</button>
+    <button class="oppia-osk-btn-operator" ng-click="$ctrl.insertString('*')">•</button>
+    <button class="oppia-osk-btn-number number-4" ng-click="$ctrl.insertString('4')">4</button>
+    <button class="oppia-osk-btn-number number-5" ng-click="$ctrl.insertString('5')">5</button>
+    <button class="oppia-osk-btn-number number-6" ng-click="$ctrl.insertString('6')">6</button>
+    <button class="oppia-osk-btn-operator" ng-click="$ctrl.insertString('-')">-</button>
+    <img ng-src="<[$ctrl.getStaticImageUrl('/icons/osk-fraction.svg')]>" class="oppia-osk-btn-operator" alt="÷" ng-click="$ctrl.insertSymbol('frac')">
```

Comment:
That would make the code quite messy, IMO. Because in that case, we would have to use 2 horizontal flexboxes, with the first one containing numbers from 9-1 and the operators and shortcuts...and the second row would contain `. 0 backspace < >`. Now, in both of these flexboxes, we would have to use 2 grids, for the left part and the right part. So, that would be a total of 4 grids with dims as: 3x3, 3x2, 1x3, 1x2. Does that not seem more complex than the current implementation which is just a single grid of dim 4x5?